sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java

```diff
@@ -539,15 +542,36 @@ public void listWithMultipleKeys(HttpClient httpClient, ConfigurationServiceVers
 
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.data.appconfiguration.TestHelper#getTestParameters")
-    public void listConfigurationSettings(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
+    public void listConfigurationSettingsWithNullSelector(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
         client = getConfigurationAsyncClient(httpClient, serviceVersion);
+        String key = getKey();
+        String key2 = getKey();
+
+        // Delete all existing settings in the resource
+        final PagedFlux<ConfigurationSetting> configurationSettingPagedFlux2 = client.listConfigurationSettings(null);
+        for (ConfigurationSetting setting : configurationSettingPagedFlux2.toIterable()) {
+            StepVerifier.create(client.deleteConfigurationSettingWithResponse(setting, false))
+                .assertNext(response -> assertConfigurationEquals(setting, response))
+                .verifyComplete();
+        }
```

Comment:
If we wanted this to be fully reactive:

```suggestion
Mono<Void> deleteAllSettingsMono = client.listConfigurationSettings(null)
    .flatMap(setting -> client.deleteConfigurationSettingWithResponse(setting, false))
    .then();
    
StepVerifier.create(deleteAllSettingsMono)
   .verifyComplete();
```

This code doesn't need to worry about asserting values returned by the delete.